pkg/util/http/transport.go

```diff
@@ -126,8 +126,12 @@ func GetProxyTransportFunc(p *config.Proxy) func(*http.Request) (*url.URL, error
 		}(r)
 
 		// Print a warning if the proxy behavior would change if the new no_proxy behavior would be enabled
-		newURL, _ := proxyConfig.ProxyFunc()(r.URL)
-		if url != newURL && url != nil {
+		var newURLString string
+		if newURL, _ := proxyConfig.ProxyFunc()(r.URL); newURL != nil {
+			newURLString = newURL.String()
+		}
+
+		if url != nil && url.String() != newURLString {
 			urlString := r.URL.String()
 			NoProxyWarningMapMutex.Lock()
 			if _, ok := NoProxyWarningMap[urlString]; !ok {
```

Comment:
Unrelated to the PR's change, but unless I'm missing something the lookup on this line and the map addition on the line below are not performed with the exact same key? (`r.URL.String()` vs. `log.SanitizeURL(r.URL.String())`) Won't this create inconsistencies?

Reply:
Agreed. 
Perhaps this could be further improved by using `logSafeURL` as a key instead of the full url as I believe proxy matches only apply to the host/IP and not the path. 

Currently `datadoghq.com/foo/bar` will log the same warning as `datadoghq.com/abc/123` because it will be registered in the map twice with different paths. This could cause this message to be needlessly noisy. What do you think?